### PR TITLE
Support responsive props

### PR DESCRIPTION
**Pull request by stephenjwatkins**
Add support for responsive properties on component APIs. This doesn't change our breakpoint system only adds infrastructure for supporting a component API in a CSS Module world.

- Create SCSS/TS utilities
- Add example implementation to Placeholder
- Add ADR; update Icon spec
- Support typing a subset of our tokens
- Add SASS map tokens output
- Add type definitions to tokens
- Make consistent the singular/plural names of token groups
- Update eslint config to scope TypeScript to TS files and JavaScript to JS files

### CODE REVIEW

File: easy-ui-tokens/src/breakpoints/base.json [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
     "xs": { "value": "0px" },
     "sm": { "value": "576px" },
     "md": { "value": "768px" },
     "lg": { "value": "992px" },
     "xl": { "value": "1200px" },
-    "xxl": { "value": "1400px" },
-    "mobile": { "value": "{breakpoints.xs}" },
-    "tablet": { "value": "{breakpoints.md}" },
-    "desktop": { "value": "{breakpoints.xl}" }
+    "xxl": { "value": "1400px" }
```
Comment (stephenjwatkins) [inline easy-ui-tokens/src/breakpoints/base.json:8]: removed the mobile, tablet, desktop aliases for now. will reassess if/how to incorporate once we revisit these with design

File: easy-ui-react/src/utilities/types.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export type Namespace<
+  Set,
+  Needle extends string,
+> = Set extends `${Needle}${infer _X}` ? _X : never;
+
+// Returns types narrowed for the specified namespace in our tokens file:
+//
+// type BreakpointsAlias = TokenNamespace<"breakpoint">
+//   xs | sm | md ...
+export type TokenNamespace<Needle extends string> = Namespace<
```
Comment (stephenjwatkins) [inline easy-ui-react/src/utilities/types.ts:18]: i wonder if `TokenGroup` would be a better name 🤔 

```
type BreakpointTokenGroup = TokenGroup<"breakpoint">
```

File: documentation/decisions/007_support_responsive_props.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+### Defining in the component
+
+#### Simple value
+
+```jsx
+import { getResponsiveValue } from "../utilities/css";
+
+export function Icon({ size = "24px" }) {
+  const style = getResponsiveValue("icon", "size", size);
```
Comment (OskiTheCoder) [inline documentation/decisions/007_support_responsive_props.md:67]: really like this

File: easy-ui-react/src/styles/_responsive-props.scss [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+) {
+  // Create a mapping of breakpoints and a list of aliases from
+  // the breakpoint namespace of our design tokens
+  $bps: ();
+  $bp-aliases: ();
+  @each $key, $value in $ezui-tokens {
+    @if string.index($key, "breakpoint-") == 1 {
+      $bp-alias: string.slice($key, string.length("breakpoint-") + 1);
+      $bps: map.set($bps, $bp-alias, $value);
+      $bp-aliases: list.append($bp-aliases, $bp-alias);
```
Comment (OskiTheCoder) [inline easy-ui-react/src/styles/_responsive-props.scss:27]: pretty cool, didnt know you could do this kinda stuff in a mixin

File: easy-ui-react/src/utilities/types.ts (hunk 2 above)
Comment (OskiTheCoder) [reply easy-ui-react/src/utilities/types.ts]: good question, i think `TokenNamespace` is sensible and clear

File: documentation/decisions/007_support_responsive_props.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+## Decision Outcome
+
+We will use a breakpoint object schema for updating property values at different breakpoints. This schema will be exposed on specific component properties that need responsive support.
+
+## More Information
+
+When developing with traditional CSS, because React doesn't support setting `style` values for media queries, developers will often reach into the CSS of components to update styles for different screen sizes. This type of mechanism breaks component encapsulation, causing inconsistency and brittle code.
+
+Easy UI will offer a mechanism on the component API to change styles based on defined breakpoints. This will keep layout and paint changes confined to our screen constraints and keep style changes narrowed to what the component API offers.
```
Comment (OskiTheCoder) [inline documentation/decisions/007_support_responsive_props.md:27]: 🎉

**Review by OskiTheCoder — APPROVED** [review]
everything lgtm, this is awesome

**Pull request merged**